python/plugins/processing/algs/gdal/warp.py

```diff
@@ -51,6 +51,8 @@ class warp(GdalAlgorithm):
     DEST_SRS = 'DEST_SRS'
     METHOD = 'METHOD'
     METHOD_OPTIONS = ['near', 'bilinear', 'cubic', 'cubicspline', 'lanczos']
+    if GdalUtils.version() >= 2000000:
+        METHOD_OPTIONS.extend(['average', 'mode', 'max', 'min', 'med', 'q1', 'q3'])
```

Comment:
Looks like average and mode are even available for older versions, see
http://www.gdal.org/gdalwarp.html